docs/how_to/single_node.md.source

```diff
@@ -0,0 +1,144 @@
+# M3DB Single Node Deployment
```

Comment:
could we start a a new tree under `docs` for a bit better organisation. 

something like:

```
github.com/m3db/m3 <- top level
|
|---- docs
         |---------------- resources (or generated)
                              |------ single_node.md.source
```

That way the sources will be in one folder, the generated docs in the right place (as they are now) and we can move the tests to - for e.g. `scripts/docker-integration-tests/...`